crates/fluvio-stream-dispatcher/src/metadata/k8.rs

```diff
@@ -0,0 +1,321 @@
+use k8_client::meta_client::MetadataClient as K8MetadataClient;
+
+use anyhow::{Result, anyhow};
+use futures_util::{stream::BoxStream, StreamExt};
+use k8_client::meta_client::NameSpace as K8NameSpace;
+use tracing::{trace, debug, error};
+
+use fluvio_stream_model::{
+    k8_types::{
+        Spec as K8Spec,
+        options::{DeleteOptions, PropogationPolicy},
+        InputK8Obj, UpdateK8ObjStatus, K8Watch,
+    },
+    store::{
+        MetadataStoreList,
+        k8::{K8MetaItem, K8ExtendedSpec, K8ConvertError},
+        MetadataStoreObject, NameSpace,
+        actions::LSUpdate,
+    },
+    core::{Spec, MetadataItem},
+};
+
+use super::MetadataClient;
+
+#[async_trait::async_trait]
+impl<T: K8MetadataClient> MetadataClient<K8MetaItem> for T {
```

Comment:
IIUC, you mean instead of:
```rust
impl <T: K8MetadataClient> MetadataClient<K8MetaItem> for T
```
do
```rust
struct BoxedClient {
    inner: Box<dyn K8MetadataClient>
}

impl impl<T: K8MetadataClient> MetadataClient<K8MetaItem> for BoxedClient 
```
right?
If so, I don't see how it could help us with decoupling Fluvio from external metadata, we still need `K8MetadataClient` trait here.

Reply:
Yup, exactly.  We have following type of metadata
- Local Metadata
- K8
- ReadOnly

Currently there are config flag in the cluster manage to dealing with them.  Having dy trait will make it easier to separate out config logic.   It will be nice if it can be done here  here but not blocker.

Reply:
I still don't understand how dyn trait makes it easier.  
To briefly recap:
Local Metadata will implement only `fluvio_stream_model::MetadataClient`, not `k8_client::MetadataClient`.
K8 and ReadOnly implement `k8_client::MetadataClient`.

To bridge `fluvio_stream_model::MetadataClient` and `k8_client::MetadataClient` we have two options:
1. Dynamic dispatch - `Box<dyn k8_client::MetadataClient>`
2. Static dispatch - `impl <T: k8_client::MetadataClient> fluvio_stream_model::MetadataClient<K8MetaItem> for T`

Both are equivalent in terms of compilation time: all deps, configs, features, etc. are the same.
While at runtime `Dynamic dispatch ` does extra vtable lookup which is extra CPU work (minor work but still).